modules/mono/editor/GodotTools/GodotTools/Export/ExportPlugin.cs

```diff
@@ -343,10 +337,27 @@ private void _ExportBeginImpl(string[] features, bool isDebug, string path, long
                                             return;
                                         }
 
-                                        static bool IsSharedObject(string fileName)
+                                        bool IsSharedObject(string fileName)
                                         {
-                                            if (fileName.EndsWith(".so") || fileName.EndsWith(".a")
-                                             || fileName.EndsWith(".dex"))
+                                            if (fileName.EndsWith(".jar"))
+                                            {
+                                                // Don't export the same jar twice. Otherwise we will have conflicts.
+                                                // This can happen when exporting for multiple architectures. Dotnet
+                                                // stores the jars in .godot/mono/temp/bin/Export[Debug|Release] per
+                                                // target architecture. Jars are cpu agnostic so only 1 is needed.
+                                                var jarName = Path.GetFileName(fileName);
+                                                if (exportedJars.Contains(jarName))
+                                                {
+                                                    return false;
+                                                }
+                                                else
+                                                {
+                                                    exportedJars.Add(jarName);
+                                                    return true;
+                                                }
```

Comment:
No need to check twice, `HashSet<T>.Add` already returns a boolean indicating whether the element was added.

```suggestion
                                                return exportedJars.Add(jarName);
```

Reply:
done :)  This is much more elegant! :)